### ISSUE TITLE
Not all tests work with Python3

### ISSUE THREAD

**Issue by hillu** [log2timeline/dfdatetime#48]
Running tests with Python3 produces a number of exceptions such as this one:
```
======================================================================
ERROR: testAtMaximumDepth (registry_searcher.FindSpecTest)
Tests the AtMaximumDepth function.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/tests/registry_searcher.py", line 103, in testAtMaximumDepth
    key_path=u'HKEY_CURRENT_USER\\Software\\Microsoft')
  File "./dfwinreg/registry_searcher.py", line 46, in __init__
    if len(key_path_arguments) > 1:
TypeError: object of type 'filter' has no len()
```
... and this one:
```
======================================================================
FAIL: testSplitKeyPath (key_paths.KeyPathTest)
Tests the SplitKeyPath function.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/tests/key_paths.py", line 22, in testSplitKeyPath
    self.assertEqual(path_segments, expected_path_segments)
AssertionError: <filter object at 0x7f0a21bb17b8> != ['HKEY_CURRENT_USER', 'Software', 'Microsoft']
```

**Comment by joachimmetz** [comment]
Thx for the report, I'll have a look when time permits

Note to self switch to tox for python3 testing

**Comment by joachimmetz** [comment]
@hillu which version of Python 3 is this?

**Comment by joachimmetz** [comment]
This is a dfwinreg issue report moved issue to: https://github.com/log2timeline/dfwinreg/issues/54

**Comment by hillu** [comment]
Hi Joachim,

I was going to repost the results of failed tests after a dfwinreg update when it occurred to me that there is an embedded copy of dfwinreg. After removing that (so so apparently the updated dfwinreg that is installed in the Python3 `dist-packages` directory is used instead, the tests pass. This is on a Debian/unstable system with Python 3.5.3, by the way.

**Comment by joachimmetz** [comment]
@hillu can you provide me some context, embedded version of dfwinreg where?

**Comment by hillu** [comment]
Uh-oh, it seems that I had a major screw-up in my git repository. It seems that I had merged dfwinreg-20170529 into my dfdatetime repository which I use to work with git-buildpackage to produce the packages in Debian. This also explains the initial dfdatetime/dfwinreg confusion. Sorry for the noise.
